internal/pkg/service/buffer/storage/local/writer/volume_test.go

```diff
@@ -77,19 +77,53 @@ func TestOpenVolume_Error_VolumeFilePermissions(t *testing.T) {
 	}
 }
 
-// TestOpenVolume_DrainFile tests that the volume can be blocked for writing by a drain file.
-func TestOpenVolume_DrainFile(t *testing.T) {
+// TestOpenVolume_DrainFile_TrueFalse tests that the volume can be blocked for writing by a drain file.
+func TestOpenVolume_DrainFile_TrueFalse(t *testing.T) {
 	t.Parallel()
 	tc := newVolumeTestCase(t)
 
 	// Create an empty drain file
-	assert.NoError(t, os.WriteFile(filepath.Join(tc.VolumePath, drainFile), nil, 0o640))
+	drainFilePath := filepath.Join(tc.VolumePath, drainFile)
+	assert.NoError(t, os.WriteFile(drainFilePath, nil, 0o640))
 
 	// Type open volume
-	_, err := tc.OpenVolume()
-	if assert.Error(t, err) {
-		assert.Equal(t, `cannot open volume for writing: found "drain" file`, err.Error())
+	vol, err := tc.OpenVolume(WithWatchDrainFile(true))
+	assert.NoError(t, err)
+	assert.True(t, vol.Drained())
+
+	// Check error
+	if strings.Contains(tc.Logger.ErrorMessages(), `ERROR  cannot create FS watcher:`) {
+		t.Skipf(`too many opened inotify watchers, many tests are probably running in parallel`)
+	}
+
+	// Remove the file
+	assert.NoError(t, os.Remove(drainFilePath))
+	assert.Eventually(t, func() bool {
+		return vol.Drained() == false
+	}, time.Second, 5*time.Millisecond)
+}
+
+// TestOpenVolume_DrainFile_FalseTrue tests that the volume can be blocked for writing by a drain file.
+func TestOpenVolume_DrainFile_FalseTrue(t *testing.T) {
+	t.Parallel()
+	tc := newVolumeTestCase(t)
+
+	// Type open volume
+	vol, err := tc.OpenVolume(WithWatchDrainFile(true))
+	assert.NoError(t, err)
+	assert.False(t, vol.Drained())
+
+	// Check error
+	if strings.Contains(tc.Logger.ErrorMessages(), `ERROR  cannot create FS watcher:`) {
+		t.Skipf(`too many opened inotify watchers, many tests are probably running in parallel`)
 	}
+
```

Comment:
Ked testujem stabilitu testov, tak ich spusti `100-1000` krat opakovane a paralelne, pre tento pripad je tu tento `Skip`, `-count 10` prejde, `-count 100` uz nie, kedze testov na `drain` mame viac, a je to spolu `>128` Linux inotify limit.